src/analysis/retail/mage/fire/core/CombustionActiveTime.tsx

```diff
@@ -71,18 +77,64 @@ class CombustionActiveTime extends Analyzer {
     if (!this.selectedCombatant.hasBuff(TALENTS.COMBUSTION_TALENT.id) || !buffApply) {
       return;
     }
-    const combustDuration = event.timestamp - buffApply.timestamp;
-    this.activeTime[buffApply.timestamp] =
-      combustDuration -
-      this.alwaysBeCasting.getActiveTimeMillisecondsInWindow(buffApply.timestamp, event.timestamp);
+    this.analyzeActiveTime(buffApply, event.timestamp);
   }
 
   combustionDowntime = () => {
-    let activeTime = 0;
-    this.activeTime.forEach((c) => (activeTime += c));
-    return activeTime / 1000;
+    let active = 0;
+    this.activeTime.forEach((c) => (active += c.downtime));
+    return active / 1000;
   };
 
+  analyzeActiveTime(buffApply: ApplyBuffEvent, buffEnd: number) {
+    const cast = HasRelatedEvent(buffApply, 'SpellCast') && GetRelatedEvent(buffApply, 'SpellCast');
+    if (!cast) {
+      return;
+    }
+
+    const duration = buffEnd - buffApply.timestamp;
+    const activeTime = this.alwaysBeCasting.getActiveTimeMillisecondsInWindow(
+      buffApply.timestamp,
+      buffEnd,
+    );
+    const activePercent = activeTime / duration;
+    const downtime = duration - activeTime;
```

Comment:
in [this](http://localhost:3000/report/BNxf9VWHzjtyrDYQ/23-Mythic+Tindral+Sageswift,+Seer+of+the+Flame+-+Kill+(5:55)/Mcbaguette/standard) log the 1st `Combustion Active Time` perf check shows `106%` Active Time and `-0.78s` Downtime, potentially might want to do this to avoid those edgecases.
```suggestion
    const activePercent = Math.min(activeTime / duration, 1);
    const downtime = Math.max(duration - activeTime, 0);
```
![image](https://github.com/WoWAnalyzer/WoWAnalyzer/assets/3404958/9278a494-f36d-45e7-8cc7-6eea33cf6cf4)

Reply:
Yea there is some kind of haste buff missing and I haven't been able to figure out what it is.  The console has a bunch of errors about fire blast being used while on cd too. I'll probably do that for now until I can figure out what's missing